### ISSUE TITLE
Logfiles not linked to steps and runs in Loom web UI 

### ISSUE THREAD

**Issue by isaacliao** [StanfordBioinformatics/loom#186]
Reported by Sowmi. Currently steps and runs don't have links to their logfiles, and logfiles are displayed as one big list. This makes it hard to figure out which logfiles are associated with which steps. Need to be able to see which steps failed, and access logfiles for those steps.

**Comment by nhammond** [comment]
Implemented one solution in f841bc12dbe56b4e6614594dc563b5b02e561f50 , but not merging because it severely degrades response time.

I started serializing TaskRun and TaskRunAttempt objects under StepRuns in calls to the API. This impacts the /api/run-request/, /api/abstract-workflow-run/, and /api/task-run/ views.

After submitting 20 hello_world runs, I timed the response for /api/run-requests. With this set of changes, response time was about 24 seconds. Without these changes, it was about 6 seconds.

It looks like nested serializers are a major problem for performance. We need the client to drill down for nested objects as needed rather than return deeply nested objects.

I won't merge this commit, because I think the performance hit would be a larger problem than the difficulty in accessing log files.
